lib/MusicBrainz/Redis.pm

```diff
@@ -4,36 +4,44 @@ use Encode;
 use Moose;
 use Redis;
 
-has '_connection' => (
+has 'server' => (
     is => 'rw',
-    isa => 'Redis',
+    isa => 'Str',
 );
 
-has '_namespace' => (
+has 'database' => (
     is => 'rw',
     isa => 'Str',
 );
 
-sub BUILD {
-    my ($self, $args) = @_;
+has 'namespace' => (
+    is => 'rw',
+    isa => 'Str',
+);
 
-    $self->_connection(Redis->new(
+has '_connection' => (
+    is => 'rw',
+    isa => 'Redis',
+    lazy => 1,
+    builder => '_build_connection',
+);
+
+sub _build_connection {
+    my $self = shift;
+    my $connection = Redis->new(
         encoding => undef,
         reconnect => 5,
-        server => $args->{server},
-    ));
-
-    if (defined $args->{database}) {
-        $self->_connection->select($args->{database});
-    }
-
-    $self->_namespace($args->{namespace});
+        server => $self->server,
+    );
+    $connection->select($self->database)
+        if defined $self->database;
+    $connection;
 }
 
 sub _prepare_key {
     my ($self, $key) = @_;
 
-    encode('utf-8', $self->_namespace . $key);
+    encode('utf-8', $self->namespace . $key);
```

Comment:
The original reason I renamed those is because the previous `sub BUILD` mapped the input arguments to the underscored names 1:1 without any change, as in `$self->_namespace($args->{namespace});`. But that isn't necessary if they have the same name, so by renaming the Moose attributes to match the arguments to the constructor, I was able to remove `sub BUILD` entirely.

You are right that this appears to break `RemoveExpiredSessions` (and it should probably be in a separate commit). Actually, because the names started with an underscore, I mistakenly expected them to be private and not used by other scripts. But now that I realize they're public, I think the change makes more sense.

Reply:
I moved the `_namespace` rename to a separate commit ("Rename MusicBrainz::Redis's _namespace attribute to namespace") and mentioned why `sub BUILD` is removed in the next commit's message.